delira/utils/codecs.py

```diff
@@ -0,0 +1,539 @@
+        return getattr(importlib.import_module(type_repr["module"]),
+                       type_repr["name"])
+
+    def _decode_function(self, obj: dict) -> typing.Union[
+            types.FunctionType, types.BuiltinFunctionType]:
+        """
+        Decode function
+
+        Parameters
+        ----------
+        obj : dict
+            dict to be decoded
+
+        Returns
+        -------
+        typing.Union[types.FunctionType, types.BuiltinFunctionType]
+            decoded function
+        """
+        # decode items in dict representation
+        function_repr = self.decode(obj)
+        return getattr(importlib.import_module(function_repr["module"]),
+                       function_repr["name"])
+
+    def _decode_class(self, obj: dict) -> typing.Any:
+        """
+        Decode arbitrary object
+
+        Parameters
+        ----------
+        obj : dict
+            dict to be decoded
+
+        Returns
+        -------
+        typing.Any
+            decoded object
+        """
+        class_repr = self.decode(obj)
+        cls_type = class_repr["type"]
+        cls_dict = class_repr["dict"]
+
+        # need to create a temporary type here (which is basically a raw
+        # object, since using object directly raises
+        # "TypeError: __class__ assignment only supported for heap types
+        # or ModuleType subclasses"
+        # After a bit of research this kind of class re-creation only
+        # seems to be possible, if the intermediate class was created in
+        # python (which is not True for the object type since this is part
+        # of Python's C Core)
+        tmp_cls = type("__tmp", (), {})
+        # create instance of temporary class
+        tmp_instance = tmp_cls()
+        # change class type
+        tmp_instance.__class__ = self.decode(cls_type)
+        # update attributes of class
+        tmp_instance.__dict__.update(self.decode(cls_dict))
+        return tmp_instance
+
+    def _decode_classargs(self, obj: dict) -> typing.Any:
+        """
+        Create an object from specified class and arguments
+
+        Parameters
+        ----------
+        obj : dict
+            dictionary which representes the object. Must include `module` and
+            `name`. Can optionally include `args` and `kwargs`.
+
+        Returns
+        -------
+        typing.Any
+            decoded object
+
+        Raises
+        ------
+        TypeError
+            arguments and name must be encoded as a dict
+        """
+        classargs = self.decode(obj)
+
+        if not isinstance(classargs, dict):
+            raise TypeError("Arguments for classargs must be defined as dict.")
+
+        obj_cls = getattr(importlib.import_module(classargs["module"]),
+                          classargs["name"])
+        args = classargs.get("args", [])
+        kwargs = classargs.get("kwargs", {})
+        return obj_cls(*args, **kwargs)
+
+    def _decode_functionargs(self, obj: dict) -> typing.Any:
```

Comment:
does this also work with classes to create a class instance?

Reply:
There is a second function called `_decode_classargs` which can be used to create class instances. `_decode_functionargs` is only intended to create functions and set arguments with `partial`.